onnxruntime/core/session/provider_registration.cc

```diff
@@ -68,7 +68,7 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
 
   if (strcmp(provider_name, "QNN") == 0) {
 #if defined(USE_QNN)
-    options->provider_factories.push_back(QNNProviderFactoryCreator::Create(provider_options));
+    options->provider_factories.push_back(QNNProviderFactoryCreator::Create(provider_options, &(options->value)));
```

Comment:
We're passing in the entire session options instead of just the config we need because the user may potentially call `AppendExecutionProvider()` before setting config options. The session options will be finalized by the time the EP is created.